### PR TITLE
Knm buy

### PR DESCRIPTION
**Pull request by fenollp**
3 possible issues

1. `is_authorized` is modified so that it returns true when number was never assigned. This should be fine though, as the only way to add numbers unassigned seem to be by searching for numbers.
1. `buy` does not return charges (thus the crossbar module needed patching). Charging may not be happening.
1. carrier name & errors are shown to the user. This may break some abstraction.

### CODE REVIEW

File: applications/crossbar/src/modules_v2/cb_phone_numbers_v2.erl [hunk 1]
```diff
@@ -401,24 +401,30 @@ put(Context, Number) ->
...
-    CB = fun() -> ?MODULE:put(cb_context:set_accepting_charges(Context), Number, ?ACTIVATE) end,
-    set_response(Result, Context, CB);
+    AccountId = cb_context:account_id(Context),
+    case knm_number:buy(Number, AccountId, Options) of
+        {'error', _}=Error ->
+            set_response(Error, Context);
+        {'ok', _} ->
+            NewOptions = [{'dry_run', cb_context:accepting_charges(Context)} | Options],
+            Result = knm_number:buy(Number, AccountId, NewOptions),
+            set_response(Result, Context)
```
Comment (lazedo) [inline applications/crossbar/src/modules_v2/cb_phone_numbers_v2.erl]: @fenollp this doesn't seem right to me, what about `charges` ?

Comment (fenollp) [inline applications/crossbar/src/modules_v2/cb_phone_numbers_v2.erl]: I agree & that's what the OC says. Forgot to ping @jamesaimonetti about this

**Comment by fenollp** [comment]
Please @jamesaimonetti take a look at this (and top comment)

**Comment by fenollp** [comment]
@jamesaimonetti look at meeee

**Comment by fenollp** [comment]
Had an epiphany. Only point **3** is relevant now.

**Comment by lazedo** [comment]
@fenollp you're almost there, let me know when (3) is solved

**Comment by fenollp** [comment]
@lazedo **3** is actually a question. Should the carrier error get all the way back to the user or just be logged somewhere? What do you think @jamesaimonetti ?

**Pull request merged**